featuretools/tests/primitive_tests/test_max_consecutive.py

```diff
@@ -11,230 +12,211 @@
+        array = pd.Series([False, False, False], dtype="bool")
         assert primitive_func(array) == 3
 
-    def test_with_nan(self):
-        primitive_instance = self.primitive()
-        primitive_func = primitive_instance.get_function()
-        array = pd.Series([False, np.nan, False, True])
-        assert primitive_func(array) == 2
 
-    def test_with_nan_skipna(self):
-        primitive_instance = self.primitive(skipna=False)
+class TestMaxConsecutiveTrue:
+    def test_regular(self):
+        primitive_instance = MaxConsecutiveTrue()
         primitive_func = primitive_instance.get_function()
-        array = pd.Series([False, np.nan, False, True])
-        assert primitive_func(array) == 1
+        array = pd.Series([True, False, True, True, True, False, True], dtype="bool")
+        assert primitive_func(array) == 3
 
-    def test_all_nan(self):
-        primitive_instance = self.primitive()
+    def test_all_true(self):
+        primitive_instance = MaxConsecutiveTrue()
         primitive_func = primitive_instance.get_function()
-        array = pd.Series([np.nan, np.nan, np.nan, np.nan])
-        assert np.isnan(primitive_func(array))
+        array = pd.Series([True, True, True, True], dtype="bool")
+        assert primitive_func(array) == 4
 
-    def test_all_nan_skipna(self):
-        primitive_instance = self.primitive(skipna=True)
+    def test_all_false(self):
+        primitive_instance = MaxConsecutiveTrue()
         primitive_func = primitive_instance.get_function()
-        array = pd.Series([np.nan, np.nan, np.nan, np.nan])
-        assert np.isnan(primitive_func(array))
+        array = pd.Series([False, False, False], dtype="bool")
+        assert primitive_func(array) == 0
 
 
+@pytest.mark.parametrize("dtype", ["float64", "int64"])
 class TestMaxConsecutiveNegatives:
-    primitive = MaxConsecutiveNegatives
-
-    def test_regular(self):
-        primitive_instance = self.primitive()
+    def test_regular(self, dtype):
+        if dtype == "int64":
+            pytest.skip("floats not supported in int64")
+        primitive_instance = MaxConsecutiveNegatives()
         primitive_func = primitive_instance.get_function()
-        array = pd.Series([1.3, -3.4, -1, -4, 10, -1.7, -4.9])
+        array = pd.Series([1.3, -3.4, -1, -4, 10, -1.7, -4.9], dtype=dtype)
         assert primitive_func(array) == 3
 
-    def test_all_int(self):
-        primitive_instance = self.primitive()
+    def test_all_int(self, dtype):
+        primitive_instance = MaxConsecutiveNegatives()
         primitive_func = primitive_instance.get_function()
-        array = pd.Series([1, -1, 2, 4, -5])
+        array = pd.Series([1, -1, 2, 4, -5], dtype=dtype)
         assert primitive_func(array) == 1
 
-    def test_all_float(self):
-        primitive_instance = self.primitive()
+    def test_all_float(self, dtype):
+        if dtype == "int64":
+            pytest.skip("floats not supported in int64")
```

Comment:
floats -> nans

Reply:
No this was intentional message. I made a better one
https://github.com/alteryx/featuretools/pull/2420/commits/a014517e40ce6dde0475b6c242bfe4566b183556